### PR TITLE
fix(journal): exchange debit/credit fields on edit

### PR DESCRIPTION
**Pull request by jniles**
This commit updates the debit/credit fields to compute their values
based on the exchange rate of the day.  This makes sure the debit/credit
values are always up to date with their debit_equiv/credit_equiv
counterparts.

### CODE REVIEW

File: server/controllers/finance/journal/index.js [hunk 1]
```diff
@@ -300,10 +305,19 @@ function editTransaction(req, res, next) {
...
   const ACCOUNT_NUMBER_QUERY = 'SELECT id FROM account WHERE number = ?';
   const ENTITY_QUERY = 'SELECT uuid FROM entity_map WHERE text = ?';
   const REFERENCE_QUERY = 'SELECT uuid FROM document_map  WHERE text = ?';
+  const EXCHANGE_RATE_QUERY = `
+    SELECT ? * IF(enterprise.currency_id = ?, 1, GetExchangeRate(enterprise.id, ?, ?)) AS amount FROM enterprise
+    JOIN project ON enterprise.id = project.enterprise_id WHERE project.id = ?;
+  `;
+
+  // these are global/shared properties of the current transaction
+  const projectId = currentTransaction[0].project_id;
```
Comment (sfount) [inline server/controllers/finance/journal/index.js]: It looks like both the client and the server logic are making use of which properties are shared on each row in a transaction - potentially it would good to define these columns in a constant definition that can be used by both the client and server. (transaction shared vs. transaction unique columns).

File: server/controllers/finance/journal/index.js [hunk 2]
```diff
@@ -251,6 +253,9 @@ function editTransaction(req, res, next) {
     .then((currentTransaction) => {
       const posted = currentTransaction[0].posted;
 
+      // bind the current transaction under edit as "oldTransaction"
+      oldTransaction = currentTransaction;
```
Comment (sfount) [inline server/controllers/finance/journal/index.js]: It would be good to use one name here with a comment description of what this actually represents - throughout this code `oldTransaction` and `currentTransaction` are the same entity but it is hard to follow the logic. 

I think `currentTransaction` is misleading because it ambiguously refers to either the current journal transaction or the current edited transaction. `oldTransaction` may be more appropriate.

**Review by sfount — COMMENTED** [review]
Code LGTM. This is a good patch, simply assigning an enterprise currency amount to transactions made in foreign currencies as before was a serious bug. 

In the future it would be good to allow the user to edit the actual debit and credit columns from the editing tools - this edit could be to represent that 10,000FC was paid instead of 5,000FC - given this system the value would have to be back calculated to the enterprise currency. At least now it is possible!

File: server/controllers/finance/journal/index.js (hunk 2 above)
Comment (jniles) [reply server/controllers/finance/journal/index.js]: I'll update references to make use of `oldTransaction` and `_oldTransaction` for when it just needs to be exposed to a larger scope.

File: server/controllers/finance/journal/index.js (hunk 1 above)
Comment (jniles) [reply server/controllers/finance/journal/index.js]: Do you want these done in this PR or can it be a `// TODO`?

**Comment by jniles** [comment]
@sfount, changes made!

File: server/controllers/finance/journal/index.js (hunk 1 above)
Comment (sfount) [reply server/controllers/finance/journal/index.js]: This can be done in a separate PR, I think it can be applied to many places so should be carefully laid out. A `TODO` would be ideal

**Comment by sfount** [comment]
@jniles LGTM
